apps/entity-service/src/entities/processors/site.processor.spec.ts

```
@@ -19,21 +35,173 @@ describe("SiteProcessor", () => {
+      const siteA = await SiteFactory.create({ name: "A Site" });
+      const siteB = await SiteFactory.create({ name: "B Site" });
+      const siteC = await SiteFactory.create({ name: "C Site" });
+      await expectSites([siteA, siteB, siteC], { sort: { field: "name" } }, { sortField: "name" });
+      await expectSites([siteA, siteB, siteC], { sort: { field: "name", direction: "ASC" } }, { sortField: "name" });
+      await expectSites(
+        [siteC, siteB, siteA],
+        { sort: { field: "name", direction: "DESC" } },
+        { sortField: "name", sortUp: false }
+      );
+    });
+
+    it("sorts by project name", async () => {
+      const projectA = await ProjectFactory.create({ name: "A Project" });
+      const projectB = await ProjectFactory.create({ name: "B Project" });
+      const projectC = await ProjectFactory.create({ name: "C Project" });
+      const siteA = await SiteFactory.create({ projectId: projectA.id });
+      const siteB = await SiteFactory.create({ projectId: projectB.id });
+      const siteC = await SiteFactory.create({ projectId: projectC.id });
+      await expectSites([siteA, siteB, siteC], { sort: { field: "projectName" } }, { sortField: "projectName" });
+      await expectSites(
+        [siteC, siteB, siteA],
+        { sort: { field: "projectName", direction: "DESC" } },
+        { sortField: "projectName" }
+      );
+    });
+
+    it("should not sort by status", async () => {
+      await SiteFactory.create({ status: "approved" });
+      await SiteFactory.create({ status: "started" });
+      await SiteFactory.create({ status: "awaiting-approval" });
+      await expect(processor.findMany({ sort: { field: "status" } } as EntityQueryDto)).rejects.toThrow(
+        BadRequestException
+      );
     });
   });
 
   describe("findOne", () => {
-    it("throws", async () => {
-      await expect(processor.findOne()).rejects.toThrow();
+    it("returns the requested site", async () => {
```

Please follow descriptive naming to specify the expected behavior, like:

`should return a requested site`
 
Sometimes is good to add alternative scenario

`should throw an error when  requested site is not found`


This naming convention is better to make clarity of your test and expected behavior.

---

changed